src/common/lib/python/common_extension.c

```diff
@@ -317,6 +319,20 @@ static int PyTask_init(PyTask *self, PyObject *args, PyObject *kwds) {
     }
   }
   utarray_free(val_repr_ptrs);
+  /* Set the resource vector of the task. */
+  if (resource_vector != NULL) {
```

Comment:
I wonder if we should be checking that the resource values specified by the caller are above some minimum value. If we use 1.0 as the default number of CPUs instead of the minimum, we could specify a task as using 0 CPUs, and then we won't track the dynamic resources properly in the schedulers. Could lead to unfair task assignments, where one task is starved just because it specified a more conservative resource requirement, whereas another task specifies 0 CPUs but actually uses n.